tests/phpunit/tests/term/getTerms.php

```diff
@@ -2988,6 +2988,191 @@ public function test_with_term_slug_equal_to_string_0() {
 		$this->assertSameSets( array( $term_id ), wp_list_pluck( $found, 'term_id' ) );
 	}
 
+
+	/**
+	 * @ticket 55352
+	 *
+	 * @dataProvider data_same_term_args
+	 */
+	public function test_cache_key_generation( $args_1, $args_2 ) {
+		global $wpdb;
+		register_taxonomy( 'wptests_tax', 'post', array( 'hierarchical' => true ) );
+
+		$term_id1 = self::factory()->term->create(
+			array(
+				'name'     => 'Foo',
+				'slug'     => 'Foo',
+				'taxonomy' => 'wptests_tax',
+			)
+		);
+
+		$term_id2 = self::factory()->term->create(
+			array(
+				'name'     => 'Bar',
+				'slug'     => 'bar',
+				'taxonomy' => 'wptests_tax',
+			)
+		);
+
+		$posts = self::factory()->post->create_many( 3, array( 'post_type' => 'post' ) );
+		foreach ( $posts as $i => $post ) {
+			wp_set_object_terms( $post, array( $term_id1, $term_id2 ), 'wptests_tax' );
+		}
+
+		$query1      = get_terms( $args_1 );
+		$num_queries = $wpdb->num_queries;
+		$query2      = get_terms( $args_2 );
+
+		$this->assertSame( $num_queries, $wpdb->num_queries );
```

Comment:
Let's use the cache domain to make sure the queries are in fact the same.

```suggestion
		$this->assertSame( $num_queries, $wpdb->num_queries );
		
		$query3 = get_terms( array_merge( $args_2, array( 'cache_domain' => microtime() ) ) );
		$this->assertSame( $query1, $query3 );
```

Reply:
As the term objects are not exactly the same, as we can't do this.

Reply:
> As the term objects are not exactly the same, as we can't do this.

How do they differ?

Reply:
> How do they differ?

Nevermind, I've got it.